src/utils.ts

```diff
@@ -59,6 +59,11 @@ export function getLocalStorage<T>(key: string): T | undefined {
 	return decoded
 }
 
+export function setLocalStorage<T>(key: string, value: T) {
+	const encoded = btoa(JSON.stringify(value));
+	localStorage.setItem(key, encoded);
+}
```

Comment:
The official site code uses this:

```ts
function setLocalStorage<T>(key: string, data: T): void {
    const encoded = Buffer.from(JSON.stringify(data), 'binary').toString('base64')
    localStorage.setItem(key, encoded)
}
```

Do you know if this has any difference?

Reply:
As far as I'm aware, there shouldn't be any difference in terms of the resulting data - tested it by converting it back and forth and reloading the page so that the page's code would have to grab the newly stored value.

However, if you want it to match their implementation exactly I can update it

Reply:
To my knowledge, using Buffer.from over btoa is probably an artifact of the dev that wrote it being used to NodeJS. There Buffer is a global since btoa is a browser-only thing.

Reply:
Good to hear. I just didn't want to accidentally interfere with the sites normal operation

Reply:
I need to test some more but btoa actually errors if the string contains unicode https://developer.mozilla.org/en-US/docs/Web/API/btoa#unicode_strings

<img width="912" alt="image" src="https://user-images.githubusercontent.com/1973559/210557123-3a4b7528-fc88-4f8d-ac6b-76c45db1edb0.png">

These kinds of names are pretty common in Steam accounts

Reply:
Yeah we literally just realised this on Discord - may just want to use Buffers everywhere if we have to pay the 2k lines code in bundle.js anyway

Reply:
This still needs fixing. We should use the same code as the official site 

```ts
function setLocalStorage<T>(key: string, data: T): void {
    const encoded = Buffer.from(JSON.stringify(data), 'binary').toString('base64')
    localStorage.setItem(key, encoded)
}
```